includes/compat/trait-wc-stripe-pre-orders.php

```diff
@@ -21,6 +30,18 @@ public function maybe_init_pre_orders() {
 		$this->supports[] = 'pre-orders';
 
 		add_action( 'wc_pre_orders_process_pre_order_completion_payment_' . $this->id, [ $this, 'process_pre_order_release_payment' ] );
+
+		/**
+		 * The callbacks attached below only need to be attached once. We don't need each gateway instance to have its own callback.
+		 * Therefore we only attach them once on the main `stripe` gateway and store a flag to indicate that they have been attached.
+		 */
+		if ( self::$has_attached_pre_order_integration_hooks || WC_Gateway_Stripe::ID !== $this->id ) {
+			return;
+		}
+
+		self::$has_attached_pre_order_integration_hooks = true;
```

Comment:
If there are any possibility for this to fail, I would move the filter inclusion above this (so we can be sure the hook was attached).

Reply:
👍 I think the risk is fairly low for this but happy to change the order of things here. I've changed it in: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2933/commits/c643fa3cee804cc24f75f169d1721c4d59e9efa4